src/transformers/utils/bitsandbytes.py

```diff
@@ -52,12 +52,16 @@ class `Int8Params` from `bitsandbytes`.
     if old_value.device == torch.device("meta") and device not in ["meta", torch.device("meta")] and value is None:
         raise ValueError(f"{tensor_name} is on the meta device, we need a `value` to put in on {device}.")
 
-    if is_buffer:
-        has_fp16_weights = None
+    is_4bit = False
+    is_8bit = False
+    if is_buffer or not is_bitsandbytes_available():
+        is_8bit = False
+        is_4bit = False
     else:
-        has_fp16_weights = getattr(module._parameters[tensor_name], "has_fp16_weights", None)
+        is_4bit = isinstance(module._parameters[tensor_name], bnb.nn.Params4bit)
```

Comment:
We should probably protect the import of `Params4bit` for users that uses a previous version of `bnb`